### PR TITLE
fix: cherrypick semver fix #20096  into 2.12

### PR DESCRIPTION
**Pull request by PaulSonOfLars**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

File: util/git/client_test.go [hunk 1]
```diff
@@ -173,6 +173,148 @@ func Test_ChangedFiles(t *testing.T) {
 	assert.ElementsMatch(t, []string{"README"}, changedFiles)
 }
 
+func Test_SemverTags(t *testing.T) {
+	tempDir := t.TempDir()
+
+	client, err := NewClientExt(fmt.Sprintf("file://%s", tempDir), tempDir, NopCreds{}, true, false, "", "")
```
Comment (blakepettersson) [inline util/git/client_test.go:179]: For 2.12 there seems to be a slight difference in API

```suggestion
	client, err := NewClientExt(fmt.Sprintf("file://%s", tempDir), tempDir, NopCreds{}, true, false, "")
```

**Review by blakepettersson — APPROVED** [review]
Minor issue otherwise LGTM

File: util/git/client_test.go (hunk 1 above)
Comment (PaulSonOfLars) [reply util/git/client_test.go:179]: Ah, thank you! completely missed that.

**Review by todaywasawesome — APPROVED** [review]
Thanks for resolving LGTM

**Review by pasha-codefresh — CHANGES_REQUESTED** [review]
Please fix linter issues

**Comment by PaulSonOfLars** [comment]
Not sure what the failing generated code check is about here; I don't think I've changed anything related to codegen. Let me know if there's anything in particular to run!

**Comment by blakepettersson** [comment]
@PaulSonOfLars that is expected in the `release-*` branches, so you should be good to go. @pasha-codefresh does this look good to you?

**Review by christianh814 — APPROVED** [review]

**Review by pasha-codefresh — APPROVED** [review]

**Pull request merged**